### PR TITLE
Introducing coding standards

### PR DESCRIPTION
**Pull request by marksabbath**
This PR is intended to bring the WordPress coding standards to this plugin. It also modifies a bit the file structure and sanitizes some variables.

The PR also introduces the CI using CircleCI tool.

### CODE REVIEW

**Comment by nathanrice** [comment]
If we remove the plugin header from `plugins.php` and move it to `genesis-simple-sidebars.php`, won't we break the plugin on update?

**Comment by marksabbath** [comment]
It does not break the site, however, makes the Plugin page to not understand that the plugin is active. I'll roll that back.

**Comment by dreamwhisper** [comment]
@marksabbath I'm testing characters from https://github.com/studiopress/genesis-simple-sidebars/issues/10#issuecomment-284011388 and have found some differences.

For example, adding a sidebar with `!@#$%^&*()`~` in the released version Name resulted in `gss-sidebar-3`, but in the proposed changes, it results in `gss-sidebar-Array`.

Also, if I "accidentally" add a duplicate sidebar name, it results in an appended number in the released version. If I do the same with the proposed changes, it hangs and never completes. I'm not sure if that's related to my local dev somehow so I'll continue testing.

**Comment by marksabbath** [comment]
@dreamwhisper I've pushed another commit that should resolve the hanging issue. It also should have resolved the issue while adding with the name `!@#$%^&*()~`.

Nice catch BTW, that was a tricky one :D

**Comment by dreamwhisper** [comment]
@marksabbath That is working for me.

gss

There are some differences though. `<foo val=`bar' />` was previously created as a sidebar, but now it isn't. I get `Oops! Please choose a valid Name for this sidebar`. If it exists already, I can delete the one previously created. I think this is probably ok, but it is a difference.

However, an additional problem I'm having now is that my sidebar selection will not change:
- from what the default option is if no sidebar was selected or
- from what the selection was prior to updating to the new version

It seems it may be related to the Classic editor plugin. I have that plugin installed and set up to allow me to use either editor depending on the post/page, etc.

If the page is using the classic editor, it doesn't save my selected option. However, if I choose to switch to the block editor, it does save my selected option. It is unrelated to special characters.

If I switch back to the released plugin version, I can save the selected option.

**Comment by marksabbath** [comment]
@dreamwhisper I've pushed a new commit that should fix the issue while saving the sidebar's selection.

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	},
+	"minimum-stability": "dev",
+	"prefer-stable": true,
+	"scripts": {
+		"phpcs": "phpcs --standard=WordPress --ignore=vendor/,node_modules/,assets/ --extensions=php -p ./",
+        "phpcs-compat": "phpcs --extensions=php --standard=PHPCompatibilityWP --ignore=vendor/,node_modules/,assets/ --runtime-set testVersion 5.6- -p ./",
+        "phpcbf": "phpcbf --standard=WordPress --ignore=vendor/,node_modules/,assets/ --extensions=php -p ./"
+	},
+	"support": {
+		"issues": "https://github.com/studiopress/genesis-simple-share/issues",
```
Comment (dreamwhisper) [inline composer.json]: @marksabbath oops, just saw this. 
```suggestion
		"issues": "https://github.com/studiopress/genesis-simple-sidebars/issues",
```

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+		"source": "https://github.com/studiopress/genesis-simple-share"
```
Comment (dreamwhisper) [inline composer.json]:
```suggestion
		"source": "https://github.com/studiopress/genesis-simple-sidebars"
```

File: composer.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	},
+	"config": {
+		"sort-order": true
+	},
+	"minimum-stability": "dev",
+	"prefer-stable": true,
+	"scripts": {
+		"phpcs": "phpcs --standard=WordPress --ignore=vendor/,node_modules/,assets/ --extensions=php -p ./",
+        "phpcs-compat": "phpcs --extensions=php --standard=PHPCompatibilityWP --ignore=vendor/,node_modules/,assets/ --runtime-set testVersion 5.6- -p ./",
+        "phpcbf": "phpcbf --standard=WordPress --ignore=vendor/,node_modules/,assets/ --extensions=php -p ./"
```
Comment (nathanrice) [inline composer.json]: Indentation: needs another tab on lines 25-26.

**Pull request merged**